src/html5/parser/document.rs

```diff
@@ -786,25 +794,13 @@ impl DocumentHandle {
             return Err(Error::Query("Query predicate is uninitialized".to_owned()));
         }
 
-        let mut found_ids = Vec::new();
-
-        let mut node_stack: Vec<NodeId> = Vec::new();
-        let root_id = self.get().get_root().id;
-        node_stack.push(root_id);
-
         let doc_read = self.get();
+        let mut found_ids = Vec::new();
+        let tree_iterator = TreeIterator::new(self);
 
-        while let Some(current_node_id) = node_stack.pop() {
+        for current_node_id in tree_iterator {
```

Comment:
cool side effect of using the iterator, we get this nice Python-like iteration syntax